core/src/Threads/Kokkos_ThreadsExec_base.cpp

```diff
@@ -100,8 +100,8 @@ bool ThreadsExec::spawn() {
 
   pthread_attr_t attr;
 
-  if (0 == pthread_attr_init(&attr) ||
-      0 == pthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM) ||
+  if (0 == pthread_attr_init(&attr) &&
+      0 == pthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM) &&
       0 == pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED)) {
```

Comment:
hm is this really &&? Not ||? Was it initially correct if any of these things are not set that it needs to recreated?

Reply:
Also, should `pthread_attr_destroy` only be called if the `pthread_attr_init` call succeeded?  The [POSIX spec](https://pubs.opengroup.org/onlinepubs/9699919799/) says that "[t]he behavior is undefined if the value specified by the `attr` argument to `pthread_attr_destroy()` does not refer to an initialized thread attributes object."